### ISSUE TITLE
SKU Partial Matching For Products Search Filter

### ISSUE THREAD

**Issue by simon-online** [saleor/saleor#10134]
When you search for the full SKU of a product it returns the matching product but if you search for the first part of an SKU it often doesn't match any products.

It would be really helpful if searching for part of an SKU would return all the matching products when using the products query ProductFilterInput search field.

**Comment by patrys** [comment]
The search feature is using full-text search, it's impossible for it to match parts of words while remaining accurate. If necessary, we could introduce a separate API for glob matching product SKUs and only SKUs.

**Comment by simon-online** [comment]
The reason I requested this was because my client noticed that they couldn't do a search for part of an SKU in the dashboard Products search like they are use to doing in WooCommerce which they are looking at switching away from.

Their SKUs are in parts separated by hyphens for example:
KT-INS114-DRE

If I did a partial SKU search for:
KT-INS114
it would not match any products but I just did some more testing and I found that if I replaced the hyphens with spaces in the dashboard Products search I was able to do a partial match with some of the SKU parts for example:
KT INS114

If I tell them to search for partial SKUs by replacing hyphens with spaces this should be acceptable. Fortunately their SKUs have hyphens otherwise this probably wouldn't have worked.
